### PR TITLE
feat: add temp-impl of heap

### PR DESCRIPTION
**Pull request by Naman2706**

### CODE REVIEW

**Comment by Naman2706** [comment]
**map of example from container pkg:**
image

**map of current main**
image

**Comment by Naman2706** [comment]
**Traditional**
```
cannot inline down: function too complex: cost 240 exceeds budget 80
cannot inline up: function too complex: cost 144 exceeds budget 80
cannot inline Push: function too complex: cost 183 exceeds budget 80
cannot inline Pop: function too complex: cost 251 exceeds budget 80
```
**New**
```
cannot inline (*heap).down: function too complex: cost 101 exceeds budget 80
can inline (*heap).up with cost 53
can inline (*heap).Push with cost 73
cannot inline (*heap).Pop: function too complex: cost 127 exceeds budget 80
```

File: pkg/heap/heap.go [hunk 1]
```diff
@@ -1 +1,66 @@
...
+	*a = heap(*arr)
+	n := a.Len()
+	for i := n/2 - 1; i >= 0; i-- {
+		a.down(i, n)
+	}
+	return a
+}
+
+func (h heap) Len() int           { return len(h) }
+func (h heap) Less(i, j int) bool { return h[i] < h[j] }
```
Comment (yashptel) [inline pkg/heap/heap.go]: this function does not need to be exposed

File: pkg/heap/heap.go [hunk 2]
```diff
@@ -1 +1,66 @@
...
+func (h heap) Swap(i, j int)      { h[i], h[j] = h[j], h[i] }
```
Comment (yashptel) [inline pkg/heap/heap.go]: same for this. This function also does not need to be exposed

**Pull request merged**